// portal/src/components/CreateProjectModal.jsx
import React, { useState, useEffect } from 'react';
import CategorySelect from "../components/CategorySelect";
import SearchableSelect from '../components/SearchableSelect';

export default function CreateProjectModal({ isOpen, onClose }) {
    const [projectName, setProjectName] = useState("");
    const [categoryId, setCategoryId] = useState("");
    const [pic, setPic] = useState("");
    const [pms, setPms] = useState([]);

    useEffect(() => {
        if (isOpen) {
            const savedUsers = localStorage.getItem('users');
            if (savedUsers) {
                const parsedUsers = JSON.parse(savedUsers);
                const projectManagers = parsedUsers.filter(u => u.role === 'Project Manager' && u.status === 'Active');
                setPms(projectManagers);
            }
        }
    }, [isOpen]);

    if (!isOpen) return null;

    const handleSubmit = (e) => {
        e.preventDefault();

        const cityObj = document.getElementById('city');
        const districtObj = document.getElementById('district');
        const clientObj = document.getElementById('client_search');

        // Generate a simple numeric ID based on timestamp
        const randomId = Date.now().toString().slice(-4);

        const newProject = {
            id: randomId,
            name: `${randomId} - ${projectName}`,
            categoryId: Number(categoryId),
            location: `${districtObj?.value || ''}, ${cityObj?.value || ''}`.replace(/^, | , $/g, '') || 'Lokasi Belum Ditentukan',
            client: clientObj?.value || 'Internal',
            pm: pic || 'TBD',
            status: 'Ongoing', // 'Persiapan' isn't explicitly mapped in the UI filters, so default to Ongoing
            progress: 0,
            value: 0,
            cost: 0,
            margin: 0,
            health: 'Good'
        };

        // Fetch existing, append, and save
        const savedData = localStorage.getItem('projects');
        // We will default to empty array here; Projects.jsx will handle the initial DB seeding if needed
        const currentProjects = savedData ? JSON.parse(savedData) : [];
        currentProjects.push(newProject);
        localStorage.setItem('projects', JSON.stringify(currentProjects));

        // Notify other components
        window.dispatchEvent(new Event('projectsUpdated'));

        onClose();
    };


    return (
        <>
            {/* Background Dashboard Simulation (Blurred) - Optional / Contextual */}
            {/* We use a fixed backdrop instead of the simulated background image since we are already inside the app */}

            {/* Modal Backdrop */}
            <div
                aria-hidden="true"
                className="fixed inset-0 bg-black/60 backdrop-blur-sm z-40 transition-opacity"
                onClick={onClose}
            ></div>

            {/* Modal Container */}
            <div className="fixed inset-0 z-50 flex items-center justify-center p-4 pointer-events-none">
                <div className="relative w-full max-w-2xl bg-white dark:bg-surface-dark rounded-xl shadow-2xl border border-gray-200 dark:border-gray-700 flex flex-col max-h-[90vh] overflow-hidden transform transition-all pointer-events-auto">

                    {/* Modal Header */}
                    <div className="px-6 py-5 border-b border-gray-100 dark:border-gray-700 flex justify-between items-center bg-white dark:bg-surface-dark sticky top-0">
                        <div>
                            <h2 className="text-xl font-semibold text-gray-900 dark:text-white flex items-center gap-2">
                                <span className="material-icons-round text-primary text-2xl">add_business</span>
                                Buat Proyek Baru
                            </h2>
                            <p className="text-sm text-gray-500 dark:text-gray-400 mt-1">Masukkan detail awal untuk memulai proyek konstruksi baru.</p>
                        </div>
                        <button
                            onClick={onClose}
                            className="text-gray-400 hover:text-gray-600 dark:hover:text-gray-200 transition-colors p-2 rounded-lg hover:bg-gray-100 dark:hover:bg-gray-700/50"
                        >
                            <span className="material-icons-round">close</span>
                        </button>
                    </div>

                    {/* Modal Body (Scrollable) */}
                    <div className="p-6 overflow-y-auto custom-scrollbar space-y-8">

                        {/* Section: Informasi Utama */}
                        <div className="space-y-4">
                            <div className="flex items-center gap-2 mb-2">
                                <span className="h-6 w-1 bg-primary rounded-full"></span>
                                <h3 className="text-sm font-semibold uppercase tracking-wider text-gray-500 dark:text-gray-400">Informasi Utama</h3>
                            </div>

                            {/* Nama Proyek */}
                            <div>
                                <label className="block text-sm font-medium text-gray-700 dark:text-gray-300 mb-1.5" htmlFor="project_name">Nama Proyek <span className="text-red-500">*</span></label>
                                <div className="relative">
                                    <span className="absolute inset-y-0 left-0 pl-3 flex items-center pointer-events-none text-gray-400">
                                        <span className="material-icons-round text-lg">edit</span>
                                    </span>
                                    <input
                                        className="pl-10 block w-full rounded-lg border-gray-300 dark:border-gray-600 bg-gray-50 dark:bg-surface-dark-lighter text-gray-900 dark:text-white shadow-sm focus:border-primary focus:ring-primary sm:text-sm py-2.5 transition-shadow"
                                        id="project_name"
                                        name="project_name"
                                        placeholder="Contoh: Renovasi Kantor Pusat PT. Maju Jaya"
                                        type="text"
                                    />
                                </div>
                            </div>



                            {/* Lokasi Grid */}
                            <div className="grid grid-cols-1 md:grid-cols-2 gap-4">
                                {/* Kota/Kabupaten */}
                                <div>
                                    <label className="block text-sm font-medium text-gray-700 dark:text-gray-300 mb-1.5" htmlFor="city">Kota / Kabupaten <span className="text-red-500">*</span></label>
                                    <div className="relative">
                                        <span className="absolute inset-y-0 left-0 pl-3 flex items-center pointer-events-none text-gray-400">
                                            <span className="material-icons-round text-lg">location_city</span>
                                        </span>
                                        <select
                                            className="pl-10 block w-full rounded-lg border-gray-300 dark:border-gray-600 bg-gray-50 dark:bg-surface-dark-lighter text-gray-900 dark:text-white shadow-sm focus:border-primary focus:ring-primary sm:text-sm py-2.5 appearance-none"
                                            id="city"
                                            name="city"
                                            defaultValue=""
                                        >
                                            <option disabled value="">Pilih Kota/Kabupaten</option>
                                            <option>Jakarta Selatan</option>
                                            <option>Jakarta Pusat</option>
                                            <option>Surabaya</option>
                                            <option>Bandung</option>
                                        </select>
                                        <span className="absolute inset-y-0 right-0 pr-3 flex items-center pointer-events-none text-gray-400">
                                            <span className="material-icons-round">expand_more</span>
                                        </span>
                                    </div>
                                </div>

                                {/* Kecamatan */}
                                <div>
                                    <label className="block text-sm font-medium text-gray-700 dark:text-gray-300 mb-1.5" htmlFor="district">Kecamatan <span className="text-red-500">*</span></label>
                                    <div className="relative">
                                        <span className="absolute inset-y-0 left-0 pl-3 flex items-center pointer-events-none text-gray-400">
                                            <span className="material-icons-round text-lg">map</span>
                                        </span>
                                        <select
                                            className="pl-10 block w-full rounded-lg border-gray-300 dark:border-gray-600 bg-gray-50 dark:bg-surface-dark-lighter text-gray-900 dark:text-white shadow-sm focus:border-primary focus:ring-primary sm:text-sm py-2.5 appearance-none"
                                            id="district"
                                            name="district"
                                            defaultValue=""
                                        >
                                            <option disabled value="">Pilih Kecamatan</option>
                                            <option>Kebayoran Baru</option>
                                            <option>Cilandak</option>
                                            <option>Tebet</option>
                                            <option>Setiabudi</option>
                                        </select>
                                        <span className="absolute inset-y-0 right-0 pr-3 flex items-center pointer-events-none text-gray-400">
                                            <span className="material-icons-round">expand_more</span>
                                        </span>
                                    </div>
                                </div>
                            </div>
                        </div>

                        {/* Separator */}
                        <hr className="border-gray-100 dark:border-gray-700" />

                        {/* Section: Tanggung Jawab & Relasi */}
                        <div className="space-y-4">
                            <div className="flex items-center gap-2 mb-2">
                                <span className="h-6 w-1 bg-primary rounded-full"></span>
                                <h3 className="text-sm font-semibold uppercase tracking-wider text-gray-500 dark:text-gray-400">Tim & Klien</h3>
                            </div>

                            {/* Cari Klien */}
                            <div>
                                <label className="block text-sm font-medium text-gray-700 dark:text-gray-300 mb-1.5" htmlFor="client_search">
                                    Klien <span className="text-xs font-normal text-gray-500 bg-gray-100 dark:bg-gray-700 px-2 py-0.5 rounded ml-2">Opsional</span>
                                </label>
                                <div className="relative group">
                                    <span className="absolute inset-y-0 left-0 pl-3 flex items-center pointer-events-none text-gray-400 group-focus-within:text-primary transition-colors">
                                        <span className="material-icons-round text-lg">search</span>
                                    </span>
                                    <input
                                        className="pl-10 block w-full rounded-lg border-gray-300 dark:border-gray-600 bg-gray-50 dark:bg-surface-dark-lighter text-gray-900 dark:text-white shadow-sm focus:border-primary focus:ring-primary sm:text-sm py-2.5"
                                        id="client_search"
                                        placeholder="Cari nama perusahaan atau klien..."
                                        type="text"
                                    />
                                </div>
                                <p className="mt-1 text-xs text-gray-500 dark:text-gray-400">Klien belum terdaftar? <a className="text-primary hover:underline" href="#">Tambah Klien Baru</a></p>
                            </div>

                            {/* Project Manager Dropdown */}
                            <div>
                                <label className="block text-sm font-medium text-gray-700 dark:text-gray-300 mb-1.5">Project Manager (PIC) <span className="text-red-500">*</span></label>
                                <SearchableSelect
                                    value={pic}
                                    onChange={(val) => setPic(val)}
                                    placeholder="Pilih Penanggung Jawab..."
                                    options={pms.map(pm => ({ value: pm.name, label: pm.name }))}
                                />
                            </div>
                        </div>
                    </div>

                    {/* Modal Footer */}
                    <div className="bg-gray-50 dark:bg-surface-dark-lighter/50 px-6 py-4 flex flex-col sm:flex-row sm:justify-end gap-3 border-t border-gray-100 dark:border-gray-700">
                        <button
                            onClick={onClose}
                            className="w-full sm:w-auto px-5 py-2.5 rounded-lg border border-gray-300 dark:border-gray-600 text-sm font-medium text-gray-700 dark:text-gray-200 hover:bg-gray-100 dark:hover:bg-gray-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-primary dark:focus:ring-offset-gray-900 transition-colors"
                            type="button"
                        >
                            Batal
                        </button>
                        <button
                            className="w-full sm:w-auto px-5 py-2.5 rounded-lg border border-transparent text-sm font-medium text-white bg-primary hover:bg-primary-hover shadow-lg shadow-primary/30 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-primary dark:focus:ring-offset-gray-900 transition-all flex items-center justify-center gap-2"
                            type="submit"
                        >
                            <span className="material-icons-round text-sm">save</span>
                            Buat Proyek
                        </button>
                    </div>
                </div>
            </div>
        </>
    );
}
